src/providers/delimitedtext/qgsdelimitedtextsourceselect.cpp

```diff
@@ -574,6 +574,10 @@ void QgsDelimitedTextSourceSelect::updateFieldLists()
     fieldNo++;
   }
 
+  // Add an empty item for M and Z field
+  cmbMField->addItem( QString() );
```

Comment:
Traditionally the blank entries would be the first ones in the list

Reply:
Updated, thanks @nyalldawson